server/lobbyconnection.py

```diff
@@ -287,8 +287,8 @@ async def send_updated_achievements(self, updated_achievements):
     async def command_admin(self, message):
         action = message['action']
 
-        if self.player.admin:
-            if action == "closeFA":
+        if action == "closeFA":
+            if await self.player_service.has_permission_role(self.player, 'ADMIN_KICK_SERVER'):
```

Comment:
You could think about moving these in the same condition i.e.
```python
if (
  action == "closeFA"
  and await self.player_service.has_permission_role(self.player, 'ADMIN_KICK_SERVER')
):
```
upside: Loses one indentation level
downside: A few more `elif` checks if it fails because the permissions are denied.
It will short-circuit though, so only ever one `has_permission_role` is awaited.